### PR TITLE
API Documentation website

### PR DESCRIPTION
**Pull request by duartepinto**
This PR includes the necessary tools and changes to allow us to deploy a website documenting this SDK. The website is hosted by Github pages and can be found at https://ios.developers.velocidi.com 
The website is generated by [jazzy](https://github.com/realm/jazzy) providing a custom template to better fit Velocidi's brand

### CODE REVIEW

File: VelocidiSDK/VelocidiSDK/Match/VSDKMatchRequest.h [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+  Class that wraps a NSURLMutableRequest with the necessary addons to make a match request to Velocidi's infrastructure
```
Comment (antonio-ramadas) [inline VelocidiSDK/VelocidiSDK/Match/VSDKMatchRequest.h]: Seems more accurate to me. What do you think?

(This suggestion also applies to other places)

```suggestion
  Class that wraps a NSURLMutableRequest with the necessary addons to make a match request to Velocidi's system
```

File: VelocidiSDK/VelocidiSDK/Tracking/VSDKTrackingRequest.h [hunk 2]
```diff
@@ -5,6 +5,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+  Class that wraps a NSURLMutableRequest with the necessary addons to make a tracking request to Velocidi's infrastructure
```
Comment (antonio-ramadas) [inline VelocidiSDK/VelocidiSDK/Tracking/VSDKTrackingRequest.h]: Seems more accurate to me. What do you think?

(This suggestion also applies to other places)

```suggestion
  Class that wraps a NSURLMutableRequest with the necessary addons to make a tracking request to Velocidi's system
```

File: VelocidiSDK/VelocidiSDK/VSDKRequest.h [hunk 3]
```diff
@@ -6,6 +6,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+  Class that wraps a NSURLMutableRequest with the necessary addons to make a request to Velocidi's infrastructure
```
Comment (antonio-ramadas) [inline VelocidiSDK/VelocidiSDK/VSDKRequest.h]: Seems more accurate to me. What do you think?

(This suggestion also applies to other places)

```suggestion
  Class that wraps a NSURLMutableRequest with the necessary addons to make a request to Velocidi's system
```

File: VelocidiSDK/VelocidiSDK/VSDKVelocidi.h [hunk 4]
```diff
@@ -7,6 +7,9 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+  VelocidiSDK main class responsible for orchestrating all the logic.
```
Comment (antonio-ramadas) [inline VelocidiSDK/VelocidiSDK/VSDKVelocidi.h]: Just for the sake of consistency.

```suggestion
  VelocidiSDK main class responsible for orchestrating all the logic
```

File: deploy.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+VERSION=$1
+DOCS_DIR=docs/docs_page
+
+#clean and copy images
+if [ -d "$DOCS_DIR" ]; then rm -Rf $DOCS_DIR; fi
+mkdir -p $DOCS_DIR/docs/img/
+cp docs/img/* $DOCS_DIR/docs/img/
+
+# build
```
Comment (antonio-ramadas) [inline deploy.sh]: What do you think about displaying an error message when jazzy is not installed?

```patch
diff --git a/deploy.sh b/deploy.sh
index 1e6442a..fe01b6d 100755
--- a/deploy.sh
+++ b/deploy.sh
@@ -17,7 +17,11 @@ if [ -d "$DOCS_DIR" ]; then rm -Rf $DOCS_DIR; fi
 mkdir -p $DOCS_DIR/docs/img/
 cp docs/img/* $DOCS_DIR/docs/img/
 
-# build
+# build docs
+if ! [ -x "$(command -v jazzy)" ]; then
+  echo 'Error: jazzy is not installed. You can install it with `gem install jazzy`.' >&2
+  exit 1
+fi
 
 jazzy \
   --objc \
```

**Review by antonio-ramadas — CHANGES_REQUESTED** [review]
Overall, it looks pretty good! I just left minor suggestions.

**Comment by antonio-ramadas** [comment]
I forgot to mention in the review, but shouldn't the README be simplified now?

File: deploy.sh (hunk 5 above)
Comment (duartepinto) [reply deploy.sh]: Also added a warning if jazzy is not in version 0.10.0 (we might have template incompatibilities). See 48243a3

**Comment by duartepinto** [comment]
> I forgot to mention in the review, but shouldn't the README be simplified now?

No, that is part of a different task

**Review by antonio-ramadas — APPROVED** [review]

**Pull request merged**